### PR TITLE
Set the forward flags on the nginx example config

### PR DESCRIPTION
**Pull request by s0undt3ch**
Refs #3300

http://trac.buildbot.net/wiki/Development
And especially:
http://trac.buildbot.net/wiki/SubmittingPatches

## Contributor Checklist:

* [ ] I have updated the unit tests
* [ ] I have created a file in the master/buildbot/newsfragment directory (and read the README.txt in that directory)
* [x] I have updated the appropriate documentation

### CODE REVIEW

File: master/docs/manual/cfg-www.rst [hunk 1]
```diff
@@ -543,20 +543,32 @@ Here is an nginx configuration that is known to work (nginx 1.6.2):
             # you could use / if you use domain based proxy instead of path based proxy
             location /buildbot/ {
                 proxy_pass http://127.0.0.1:5000/;
+                proxy_set_header Host $host;
```
Comment (tardyp) [inline master/docs/manual/cfg-www.rst]: I dont think all this complexity is even needed.
Here is what we are using in the nine.buildbot.net

https://github.com/buildbot/buildbot-infra/blob/master/roles/nginx/templates/proxy

I think the whole think could be changed as

    proxy_set_header HOST $host;
    proxy_set_header X-Real-IP $remote_addr;
    proxy_set_header X-Forwarded-For $proxy_add_x_forwarded_for;
    proxy_set_header X-Forwarded-Proto  $scheme;
    proxy_set_header X-Forwarded-Server  $host;
    proxy_set_header X-Forwarded-Host  $host;
    map $http_upgrade $connection_upgrade {
        default upgrade;
        ''      close;
    }
    location /buildbot/ {
        proxy_pass http://127.0.0.1:5000/;
        proxy_buffering off;
        proxy_read_timeout 600;

        # These three lines are be required for web socket proxying
        proxy_http_version 1.1;
        proxy_set_header Upgrade $http_upgrade;
        proxy_set_header Connection $connection_upgrade;
    }

**Comment by s0undt3ch** [comment]
Updated docs with your suggestion.

As a side note, I continue to be able to login/loout against GitHub using chromium, but not Firefox(this is using the above nginx setup).

**Comment by tardyp** [comment]
@s0undt3ch do you have more info on what is different with FF?

can you compare with https://nine.buildbot.net/#/ which is also implementing github auth?

**Comment by s0undt3ch** [comment]
My guess is something related to the websocket implementation being different on these browsers?

I previously logged in into your nine setup(can't remember which browser I used) successfully but today either using the desktop Firefox or the mobile Firefox I'm unable.

My previous CI process used Jenkins which was also linked to github authentication and I never ran into problems but I believe Jenkins is not using websockets, it's polling.

You can test this behavior by using Firefox, the first login attempt will be successful. Logout and then try to login again.

I'm using Firefox 53.0.2, at least the mobile version of it.

**Comment by tardyp** [comment]
@s0undt3ch Indeed, I can reproduce :-/

**Comment by s0undt3ch** [comment]
I tried looking at the UI code, but I'm to new to this code to find what I was looking for.
Does one need to be explicit about which cookies are used for the websocket connection?
Its a weird question because the websocket connection is an HTTP connection(which the browser should reuse known cookies) which then gets upgraded....

**Comment by tardyp** [comment]
@s0undt3ch see my investigation report at https://github.com/buildbot/buildbot/issues/3306 
It is pretty nasty race condition

**Pull request merged**